### PR TITLE
ovnkube-master pod failed to reconnect to ovn db due to ssl expire

### PR DESCRIPTION
**Pull request by winsopc**
When the cert/key files got removed during cert/key update,
With current code, libovsdb will only works for once with first remove event.
To fix this issue, watcher need remove the old file and add new file
back to watcher.

Signed-off-by: Zhen Wang <zhewang@nvidia.com>

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by dcbw** [comment]
@winsopc could we split the Remove and Write cases? I think this PR will no longer support changing the cert files in-place (eg, Write-ing new content, rather than unlink + create). I'm not sure if we care, but that used to work, and I don't think it will anymore.

**Comment by winsopc** [comment]
@girishmg @dcbw PTAL

File: go-controller/pkg/libovsdb/libovsdb.go [hunk 1]
```diff
@@ -191,6 +193,22 @@ func newSSLKeyPairWatcherFunc(certFile, privKeyFile string, tlsConfig *tls.Confi
...
+						if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+							if _, err := os.Stat(event.Name); os.IsNotExist(err) {
+								return false, nil
+							}
+							return true, nil
+						}); err != nil {
+							klog.Errorf("Fatal error: tiemout waiting for %s to be created", event.Name)
+							os.Exit(1)
+						}
+						if err := watcher.Add(event.Name); err != nil {
```
Comment (trozet) [inline go-controller/pkg/libovsdb/libovsdb.go:207]: as @dcbw pointed out to me, there is probably no reason to add back the event to watch the same file right?

File: go-controller/pkg/libovsdb/libovsdb.go [hunk 2]
```diff
@@ -191,6 +193,22 @@ func newSSLKeyPairWatcherFunc(certFile, privKeyFile string, tlsConfig *tls.Confi
...
 				if ok && event.Op&(fsnotify.Write|fsnotify.Remove) != 0 {
+					if event.Op&fsnotify.Remove != 0 {
+						// cert/key file removed, need wait for the file to be created again.
+						if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+							if _, err := os.Stat(event.Name); os.IsNotExist(err) {
+								return false, nil
+							}
+							return true, nil
+						}); err != nil {
+							klog.Errorf("Fatal error: tiemout waiting for %s to be created", event.Name)
```
Comment (rbbratta) [inline go-controller/pkg/libovsdb/libovsdb.go:204]: Typo `s/tiemout/timeout/`

**Pull request merged**